scripts/github_issue_retriever.py

```diff
@@ -48,14 +50,20 @@ def replace_with_private(
                     continue
 
                 owner, repo, issue_number = extracted
-                private_issue = self.github.fetch_issue_by_number(
-                    owner, repo, issue_number
-                )
-                if private_issue:
-                    item["title"] = private_issue["title"]
-                    item["body"] = private_issue["body"]
-                    updated_ids.add(item["id"])
-                    updated_issues.append(item)
+                try:
+                    private_issue = self.github.fetch_issue_by_number(
+                        owner, repo, issue_number
+                    )
+
+                    if private_issue:
+                        item["title"] = private_issue["title"]
+                        item["body"] = private_issue["body"]
+                        updated_ids.add(item["id"])
+                        updated_issues.append(item)
+
+                except requests.HTTPError as e:
+                    logger.info(e)
```

Comment:
Maybe we should re-raise the exception in case of an error that is not 404?

Reply:
Good point :) I've changed that. Also realized the error is actually "410 Gone" in this case, so I've adjusted the test as well